### PR TITLE
explicitly list analyzer exceptions when they fail a test

### PR DESCRIPTION
**Pull request by brettfo**
While investigating flaky tests I found that whoever wrote this helper method <sub>it was me</sub> didn't think it prudent to actually list the analyzer diagnostics that failed the test.  This fixes that so that flaky test investigation is hopefully easier.

@dotnet/roslyn-ide @dotnet/roslyn-infrastructure

### CODE REVIEW

**Comment by TyOverby** [comment]
:+1:

File: src/EditorFeatures/Test/Diagnostics/AbstractDiagnosticProviderBasedUserDiagnosticTest.cs [hunk 1]
```diff
@@ -75,7 +75,11 @@ public abstract class AbstractDiagnosticProviderBasedUserDiagnosticTest : Abstra
         /// </summary>
         private void AssertNoAnalyzerExceptionDiagnostics(IEnumerable<Diagnostic> diagnostics)
         {
-            Assert.Equal(0, diagnostics.Count(diag => diag.Descriptor.CustomTags.Contains(WellKnownDiagnosticTags.AnalyzerException)));
+            var analyzerExceptionDiagnostics = diagnostics.Where(diag => diag.Descriptor.CustomTags.Contains(WellKnownDiagnosticTags.AnalyzerException));
+
+            // using `Assert.True()` because it allows a custom message to help diagnose the problem
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Test/Diagnostics/AbstractDiagnosticProviderBasedUserDiagnosticTest.cs]: It sucks that Assert.Empty doesn't just list the elements. Should we just fix that on their end? Or have an AssertEx.Empty until then? I suspect we do this pattern a lot of places.

**Comment by jasonmalinowski** [comment]
:+1:

**Comment by dpoeschl** [comment]
:+1:

**Comment by brettfo** [comment]
I've added `AssertEx.Empty()` that displays the unexpected items in the list.  I also considered updating other tests but `findstr` reported 522 instances of `Assert.Empty()` and I didn't even dare search for anything like what I was doing with asserting the count was zero.

**Comment by davkean** [comment]
@dotnet-bot retest prtest/win/dbg/unit32 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit32/3181/
// Retest reason: https://github.com/dotnet/roslyn/issues/7976

**Pull request merged**